src/lib/Quiz.jsx

```diff
@@ -4,19 +4,69 @@ import defaultLocale from './Locale';
 import './styles.css';
 
 const Quiz = function ({
-  quiz, shuffle, showDefaultResult, onComplete, customResultPage,
-  showInstantFeedback, continueTillCorrect, revealAnswerOnSubmit,
-  allowNavigation, onQuestionSubmit, disableSynopsis,
+  quiz,
+  shuffle,
+  shuffleAnswer,
+  showDefaultResult,
+  onComplete,
+  customResultPage,
+  showInstantFeedback,
+  continueTillCorrect,
+  revealAnswerOnSubmit,
+  allowNavigation,
+  onQuestionSubmit,
+  disableSynopsis,
 }) {
   const [start, setStart] = useState(false);
   const [questions, setQuestions] = useState(quiz.questions);
-  const nrOfQuestions = (quiz.nrOfQuestions && quiz.nrOfQuestions < quiz.questions.length) ? quiz.nrOfQuestions : quiz.questions.length;
-
+  const nrOfQuestions = quiz.nrOfQuestions && quiz.nrOfQuestions < quiz.questions.length
+    ? quiz.nrOfQuestions
+    : quiz.questions.length;
+
+  // Shuffle answers funtion here
+  const shuffleAnswerSequence = (oldQuestions = []) => {
+    const newQuestions = oldQuestions.map((question) => {
+      const answerWithIndex = question.answers?.map((ans, i) => [ans, i]);
+      const shuffledAnswersWithIndex = answerWithIndex.sort(
+        () => Math.random() - 0.5,
+      );
+      const shuffledAnswers = shuffledAnswersWithIndex.map((ans) => ans[0]);
+      if (question.answerSelectionType === 'single') {
+        const oldCorrectAnswer = question.correctAnswer;
+        const newCorrectAnswer = shuffledAnswersWithIndex.findIndex(
+          (ans) => `${ans[1] + 1}` === `${oldCorrectAnswer}`,
+        ) + 1;
+        return {
+          ...question,
+          correctAnswer: `${newCorrectAnswer}`,
+          answers: shuffledAnswers,
+        };
+      }
+      if (question.answerSelectionType === 'multiple') {
+        const oldCorrectAnswer = question.correctAnswer;
+        const newCorrectAnswer = oldCorrectAnswer.map(
+          (cans) => shuffledAnswersWithIndex.findIndex(
+            (ans) => `${ans[1] + 1}` === `${cans}`,
+          ) + 1,
+        );
+        return {
+          ...question,
+          correctAnswer: newCorrectAnswer,
+          answers: shuffledAnswers,
+        };
+      }
+      return question;
+    });
+    return newQuestions;
+  };
   const shuffleQuestions = useCallback((q) => {
     for (let i = q.length - 1; i > 0; i -= 1) {
       const j = Math.floor(Math.random() * (i + 1));
       [q[i], q[j]] = [q[j], q[i]];
     }
+    if (shuffleAnswer) {
+      q = shuffleAnswerSequence(q);
+    }
```

Comment:
shuffleAnswer shouldn't depend on shuffleQuestion. In other words, I should be able to just shuffle answer without shuffling questions. This shouldn't placed inside `shuffleQuestions`

Reply:
ok, i got it, let me do it